engine/apps/api/views/user.py

```diff
@@ -292,11 +297,29 @@ def timezone_options(self, request):
         throttle_classes=[GetPhoneVerificationCodeThrottlerPerUser, GetPhoneVerificationCodeThrottlerPerOrg],
     )
     def get_verification_code(self, request, pk):
+        """
+        See `DRF_RECAPTCHA_TESTING` in `settings/base.py`
+        and [here](https://github.com/llybin/drf-recaptcha#testing) to better understand
+        when the recaptcha checks are actually made
+        """
+        logger.info("Validating reCAPTCHA code")
+
+        serializer = MobileVerificationCodeRecaptchaSerializer(
+            data={"recaptcha": request.headers.get("X-OnCall-Recaptcha", "some-non-null-value")},
+            context={"request": request},
+        )
+
+        if not serializer.is_valid():
```

Comment:
since `DRF_RECAPTCHA_TESTING` is set to `True` in `settings/base.py` this will always return true (without the default of `some-non-null-value` this would complain about the serializer field being blank).

When `DRF_RECAPTCHA_TESTING` is set to `False` (ex. for cloud stacks), `drf_django` will make actual calls to the Google reCAPTCHA server with the value of `X-OnCall-Recaptcha` (and thus `some-non-null-value` would fail).